nixos/modules/services/security/fail2ban.nix

```diff
@@ -188,46 +194,74 @@ in
         '';
       };
 
-      daemonConfig = mkOption {
-        default = ''
-          [Definition]
-          logtarget = SYSLOG
-          socket    = /run/fail2ban/fail2ban.sock
-          pidfile   = /run/fail2ban/fail2ban.pid
-          dbfile    = /var/lib/fail2ban/fail2ban.sqlite3
-        '';
-        type = types.lines;
+      daemonSettings = mkOption {
+        default = {
+          Definition = {
+            logtarget = "SYSLOG";
+            socket = "/run/fail2ban/fail2ban.sock";
+            pidfile = "/run/fail2ban/fail2ban.pid";
+            dbfile = "/var/lib/fail2ban/fail2ban.sqlite3";
+          };
```

Comment:
This should probably be more granular, i.e.
```nix
config = {
  services.fail2ban.daemonSettings.Definition = {
    logtarget = mkDefault "SYSLOG";
    # ...
  };
};
```
and leave this as a `defaultText` for informational purposes.